sycl/include/sycl/detail/nd_range_view.hpp

```diff
@@ -29,6 +29,10 @@ class nd_range_view {
   nd_range_view &operator=(const nd_range_view &Desc) = default;
   nd_range_view &operator=(nd_range_view &&Desc) = default;
 
+  template <int Dims_>
```

Comment:
shouldn't these be templated with `<size_t Dims_>` instead of `int`  ?   Then we won't need the cast.

Reply:
+1. And also `nd_range_view` implies view of `sycl::nd_range`, not `sycl::range`. I think we should have an anologous `range_view` as well.